src/compass/s1_geocode_slc.py

```diff
@@ -133,44 +147,74 @@ def run(cfg: GeoRunConfig):
+                                   lines_per_block, cols_per_block,
+                                   geogrid_expansion_threshold):
+
+                # extract start and size from slice
+                start_col, width = _slice_to_start_and_size(rdr_blk_slice[0])
+                start_line, length = _slice_to_start_and_size(rdr_blk_slice[1])
+
+                # Build input arrays to be passed to geocode_slc
+                rdr_data_blks = []
+                geo_data_blks = []
+                for rdr_dataset in rdr_datasets:
+                    rdr_data_blks.append(
+                        rdr_dataset.ReadAsArray(start_line, start_col,
+                                                length, width))
+
+                    # Init output blocks/arrays lists to NaN
+                    geo_data_blks.append(np.zeros(geo_blk_shape,
+                                                  dtype=np.complex64))
 
                 # Geocode
-                isce3.geocode.geocode_slc(geo_burst_raster, rdr_burst_raster,
-                                          dem_raster,
-                                          radar_grid, sliced_radar_grid,
-                                          geo_grid, orbit,
-                                          native_doppler,
-                                          image_grid_doppler, ellipsoid, threshold,
-                                          iters, blocksize, flatten,
-                                          azimuth_carrier=az_carrier_poly2d,
+                isce3.geocode.geocode_slc(geo_data_blks, rdr_data_blks,
+                                          dem_raster, radar_grid,
+                                          sliced_radar_grid, blk_geo_grid,
+                                          orbit, native_doppler,
+                                          image_grid_doppler, ellipsoid,
+                                          threshold, iters, start_col,
+                                          start_line, flatten,
+                                          az_carrier=az_carrier_poly2d,
                                           az_time_correction=az_lut,
                                           srange_correction=rg_lut)
 
+                # write geocoded blocks to respective HDF5 datasets
+                for geo_dataset, geo_data_blk in zip(geo_datasets,
+                                                     geo_data_blks):
+                    # write to GSLC block HDF5
+                    geo_dataset.write_direct(geo_data_blk,
+                                             dest_sel=geo_blk_slice)
+
             # Set geo transformation
+            '''
             geotransform = [geo_grid.start_x, geo_grid.spacing_x, 0,
                             geo_grid.start_y, 0, geo_grid.spacing_y]
             geo_burst_raster.set_geotransform(geotransform)
             geo_burst_raster.set_epsg(epsg)
             del geo_burst_raster
+            '''
```

Comment:
How do we assign the geotransformation now that we are not using `isce3.io.Raster`?

Reply:
This block is something copied from the NISAR code that I sloppily didn't remove. We set the geotransform in [`compass.utils.h5_helpers.init_geocoded_dataset`](https://github.com/opera-adt/COMPASS/blob/1a8db1f0a365070702560e6466c88fcdd28d243e/src/compass/utils/h5_helpers.py#L80)